pkg/txn/rpc/sender.go

```diff
@@ -24,23 +24,66 @@ import (
 	"go.uber.org/zap"
 )
 
+// WithSenderPayloadBufferSize set buffer size for copy payload data to socket.
+func WithSenderPayloadBufferSize(value int) SenderOption {
+	return func(s *sender) {
+		s.options.payloadCopyBufferSize = value
+	}
+}
+
+// WithSenderBackendOptions set options for create backend connections
+func WithSenderBackendOptions(options ...morpc.BackendOption) SenderOption {
+	return func(s *sender) {
+		s.options.backendCreateOptions = options
+	}
+}
+
+// WithSenderClientOptions set options for create client
+func WithSenderClientOptions(options ...morpc.ClientOption) SenderOption {
+	return func(s *sender) {
+		s.options.clientOptions = options
+	}
+}
+
 type sender struct {
 	logger *zap.Logger
 	client morpc.RPCClient
+
+	options struct {
+		payloadCopyBufferSize int
+		backendCreateOptions  []morpc.BackendOption
+		clientOptions         []morpc.ClientOption
+	}
 }
 
 // NewSender create a txn sender
-func NewSender(logger *zap.Logger) (TxnSender, error) {
+func NewSender(logger *zap.Logger, options ...SenderOption) (TxnSender, error) {
 	logger = logutil.Adjust(logger)
-	codec := morpc.NewMessageCodec(func() morpc.Message { return &txn.TxnResponse{} }, 0)
-	bf := morpc.NewGoettyBasedBackendFactory(codec,
-		morpc.WithBackendConnectWhenCreate(),
-		morpc.WithBackendLogger(logger))
-	client, err := morpc.NewClient(bf, morpc.WithClientLogger(logger))
+	s := &sender{logger: logger}
+	for _, opt := range options {
+		opt(s)
+	}
+	s.adjust()
+
+	codec := morpc.NewMessageCodec(func() morpc.Message { return &txn.TxnResponse{} }, s.options.payloadCopyBufferSize)
+	bf := morpc.NewGoettyBasedBackendFactory(codec, s.options.backendCreateOptions...)
+	client, err := morpc.NewClient(bf, s.options.clientOptions...)
 	if err != nil {
 		return nil, err
 	}
-	return &sender{logger: logger, client: client}, nil
+	s.client = client
+	return s, nil
+}
+
+func (s *sender) adjust() {
+	if s.options.payloadCopyBufferSize == 0 {
+		s.options.payloadCopyBufferSize = 16 * 1024
```

Comment:
make this magic number a const variable with a meaningful name and a little bit comments on why it is 16Kbytes.